homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,302 @@
+    HomeAssistant,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+    StateType,
+)
+from homeassistant.const import (
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfPressure,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolume,
+    UnitOfVolumeFlowRate,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WatergateConfigEntry
+from .coordinator import WatergateAgregatedRequests, WatergateDataCoordinator
+from .entity import WatergateEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+WIFI_CONNECTION_SENSOR_NAME = "Wifi connection"
+MQTT_CONNECTION_SENSOR_NAME = "MQTT connection"
+UPTIME_SENSOR_NAME = "Uptime"
+UPTIME_SENSOR_NAME = "Uptime"
+POWER_SUPPLY_SENSOR_NAME = "Power supply"
+WATER_FLOWING_SENSOR_NAME = "Water flowing"
+PRESSURE_SENSOR_NAME = "Water pressure"
+TEMPERATURE_SENSOR_NAME = "Water temperature"
+FLOW_SENSOR_NAME = "Water flow rate"
+IP_SENSOR_NAME = "IP address"
+GATEWAY_SENSOR_NAME = "Gateway address"
+SUBNET_SENSOR_NAME = "Subnet"
+SSID_SENSOR_NAME = "SSID"
+RSSI_SENSOR_NAME = "RSSI"
+WIFI_UPTIME_SENSOR_NAME = "WIFI Uptime"
+MQTT_UPTIME_SENSOR_NAME = "MQTT Uptime"
+SHUT_OFF_EVENT_SENSOR_NAME = "Last Auto Shut off event"
+WATER_METER_VOLUME_SENSOR_NAME = "Water meter volume"
+WATER_METER_DURATION_SENSOR_NAME = "Water meter duration"
+
+ASO_EVENT_TYPE_ATRIBUTE = "type"
+ASO_EVENT_VOLUME_ATRIBUTE = "volume"
+ASO_EVENT_DURATION_ATRIBUTE = "duration"
+
+
+WIFI_CONNECTION_ENTITY_NAME = "wifi_status"
+MQTT_CONNECTION_ENTITY_NAME = "mqtt_status"
+UPTIME_ENTITY_NAME = "uptime"
+POWER_SUPPLY_ENTITY_NAME = "power_supply"
+WATER_FLOWING_ENTITY_NAME = "water_flowing"
+PRESSURE_ENTITY_NAME = "water_pressure"
+TEMPERATURE_ENTITY_NAME = "water_temperature"
+FLOW_ENTITY_NAME = "water_flow_rate"
+IP_ENTITY_NAME = "ip"
+GATEWAY_ENTITY_NAME = "gateway"
+SUBNET_ENTITY_NAME = "subnet"
+SSID_ENTITY_NAME = "ssid"
+RSSI_ENTITY_NAME = "rssi"
+WIFI_UPTIME_ENTITY_NAME = "wifi_uptime"
+MQTT_UPTIME_ENTITY_NAME = "mqtt_uptime"
+SHUT_OFF_EVENT_ENTITY_NAME = "auto_shut_off_event"
+WATER_METER_VOLUME_ENTITY_NAME = "water_meter_volume"
+WATER_METER_DURATION_ENTITY_NAME = "water_meter_duration"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WatergateConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all entries for Watergate Platform."""
+
+    coordinator = config_entry.runtime_data
+
+    entities: list[SensorEntity] = [
+        SonicSensor(
+            coordinator,
+            WATER_METER_VOLUME_SENSOR_NAME,
+            WATER_METER_VOLUME_ENTITY_NAME,
+            lambda data: data.state.water_meter.volume
+            if data.state and data.state.water_meter
+            else None,
+            unit=UnitOfVolume.LITERS,
+            device_class=SensorDeviceClass.WATER,
+            state_class=SensorStateClass.TOTAL,
+        ),
```

Comment:
You could hand in id, translations_id, unit, *_class as [entity-description](https://developers.home-assistant.io/docs/core/entity/#entity-description)